### ISSUE TITLE
Cluster jobs fail claiming mamba is missing, even though mamba is present, and even if conda backend is specified

### ISSUE THREAD

**Issue by BEFH** [snakemake/snakemake#1009]
**Snakemake version**
6.4.0 (not present in 6.3.0)

**Describe the bug**
Cluster jobs fail with an error saying mamba is not found. Mamba works for the workflow up until submission, and the error is still present when using `--conda-frontend conda`. I am using FISH as my main shell, but the rules are running in BASH. Both my `config.fish` and `.bashrc` load conda envs that have mamba.

**Logs**

The following error shows up in the cluster log:

```
Error: mamba package manager is not available. The mamba package manager (https://github.com/mamba-org/mamba) is an extremely fast and robust conda replacement. It is the recommended way of using Snakemake's conda integration. It can be installed with `conda install -n base -c conda-forge mamba.If you still prefer to use conda, you can enforce that by setting `--conda-frontend conda`.
```

**Minimal example**
Mamba works for generating the environments, but it breaks when it tries to run the rules. I'm using the following command: ` snakejob -pj 600 --use-conda`

snakejob looks like this:

```
#!/usr/bin/env bash

set -e

if [ ! -d ".snakejob" ]; then
  mkdir .snakejob
fi

curdir="$(pwd)"
jname="$(basename $curdir)"

bsub=("bsub -K -J $jname:{rule}:{wildcards}"
"-P {cluster.project} -q {cluster.queue}"
"-n {cluster.cores} -R \"span[hosts=1] select[mem>{cluster.mem}]"
"rusage[mem={cluster.mem}]\" -W {cluster.time} -L /bin/bash"
"-oo .snakejob/{rule}:{wildcards}.stdout"
"-eo .snakejob/{rule}:{wildcards}.stderr < ")

snakemake -u cluster.yaml --cluster-sync "${bsub[*]}" \
  --local-cores 4 --max-jobs-per-second 5 "$@"
```

It's basically a script that allows me to use a cluster config file on lsf. I know this behavior is technically deprecated, but we have not been able to implement profiles to our satisfaction.

**Comment by mbhall88** [comment]
Also, as an aside @BEFH - there is an LSF profile already https://github.com/Snakemake-Profiles/lsf

**Comment by BEFH** [comment]
Is there an update on this bug? The LSF profile does not work with our cluster.

**Comment by mbhall88** [comment]
> Is there an update on this bug? The LSF profile does not work with our cluster.

Please raise an issue on the LSF profile if it isn't working on your cluster, all bugs are very much appreciated.

Regarding this issue, could you provide an example of one of the rules that uses a conda environment that is failing to run?

**Comment by BEFH** [comment]
I fixed this by switching from the anaconda base provided by the cluster admins to my own installation. My theory is that it was due to a problem with incorrect order of paths in the PATH variable. Snakemake *shouldn't* be susceptible to that since it supposedly passes the conda path (and I confirmed this in the cluster logs), but I have no way of properly debugging this at this point.

It happened both with bash shell rules with conda environments and without, as long as --use-conda was specified.

It also happened on Snakemake 6.7.0 and 6.8.0.

**Comment by slambrechts** [comment]
I'm having the same problem. When I run  `snakemake --use-conda -k -j 8` I get:

`Error: mamba package manager is not available. .....`

and when I run 

`snakemake --use-conda -k -j 8 --conda-frontend conda
`
I get:

```
CreateCondaEnvironmentException:
The 'conda' command is not available in the shell /usr/bin/bash that will be used by Snakemake. You have to ensure that it is in your PATH, e.g., first activating the conda base environment with `conda activate base`.
```

outside of snakemake, mamba and conda work just fine, so they are definitely available. Snakemake for some reason does not find them?

**Comment by BEFH** [comment]
@slambrechts, what shell are you using as your login shell and what are you using as your Snakemake shell? That might have something to do with it.

**Comment by beardymcjohnface** [comment]
I'm getting the same issue. The fix from #1019 only works if mamba is installed on the base environment.
